cli/commands.go

```diff
@@ -1,45 +1,97 @@
 package cli
 
 import (
+	"errors"
 	"fmt"
-	"os"
+	"regexp"
 	"strings"
 
 	"github.com/rokka-io/rokka-go/rokka"
 )
 
+var positionalArgsRegexp *regexp.Regexp
+
+func init() {
+	positionalArgsRegexp = regexp.MustCompile("^<(.*)>$")
+}
+
+type UnknownCommandError string
+
+func (e UnknownCommandError) Error() string {
+	return string(e)
+}
+
 type Command struct {
-	args        []string
-	description string
-	fn          func(*rokka.Client)
+	Args        []string
+	Options     []string
+	Description string
+	fn          func(*rokka.Client, map[string]string, map[string]string) error
 }
 
-var commands = []Command{
-	Command{[]string{"stackoptions", "list"}, "Show default stack options", func(c *rokka.Client) {
-		res, err := c.GetStackoptions()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error getting stack options: %s", err)
-			os.Exit(1)
+func (c *Command) TakesOption(key string) bool {
+	for _, v := range c.Options {
+		if v == key {
+			return true
 		}
-		PrettyPrintJSON(res)
-	}},
+	}
+
+	return false
 }
 
-func GetCommands() []Command {
-	return commands
+var Commands = []Command{
+	Command{[]string{"stackoptions", "list"}, nil, "Show default stack options", getStackoptions},
+	Command{[]string{"organizations", "get", "<name>"}, nil, "Get details of an organization", getOrganization},
 }
 
-func ExecCommand(c *rokka.Client, args []string) {
-	for _, v := range commands {
-		// TODO: improve this hacky line
-		if strings.Join(v.args, " ") == strings.Join(args, " ") {
-			v.fn(c)
-			return
+func ExecCommand(cl *rokka.Client, userArgs []string) error {
+	hasMatch := false
+
+	for _, c := range Commands {
+		commandArgsCount := len(c.Args)
+
+		if len(userArgs) < commandArgsCount {
+			continue
+		}
+
+		positionalArgs := make(map[string]string)
+
+		for i, arg := range c.Args {
+			// check whether this is a positional argument ("<arg>")
+			positionalMatch := positionalArgsRegexp.FindStringSubmatch(arg)
+
+			if len(positionalMatch) == 2 {
+				positionalArgs[positionalMatch[1]] = userArgs[i]
+			} else if arg != userArgs[i] {
+				// user provided argument doesn't match
+				break
+			}
+
+			// we reached the end
+			if commandArgsCount == i+1 {
+				hasMatch = true
+			}
+		}
+
+		if hasMatch {
+			options := make(map[string]string)
+
+			// parse rest into options ("key=value")
+			if len(userArgs) > commandArgsCount {
+				for _, option := range userArgs[commandArgsCount-1:] {
+					split := strings.Split(option, "=")
+					if len(split) == 2 && split[0] != "" && split[1] != "" {
+						if !c.TakesOption(split[0]) {
+							return errors.New(fmt.Sprintf(`Unsupported option "%s" for command "%s"`, split[0], strings.Join(userArgs[:commandArgsCount], " ")))
```

Comment:
you can use [`fmt.Errorf(format string, a ...interface{}) error`](https://godoc.org/fmt#Errorf) for cases like these.